src/NBitcoin/Network.cs

```diff
@@ -10,1429 +12,1079 @@
+    {
+        /// <summary>
+        /// Height in coinbase
+        /// </summary>
+        BIP34,
+        /// <summary>
+        /// Height in OP_CLTV
+        /// </summary>
+        BIP65,
+        /// <summary>
+        /// Strict DER signature
+        /// </summary>
+        BIP66
+    }
+
+    public class Consensus
+    {
+        /// <summary>
+        /// An extension to <see cref="Consensus"/> to enable additional options to the consensus data.
+        /// </summary>
+        public class ConsensusOptions
+        {
+        }
+
+        public ConsensusOptions Options { get; set; }
+
+        public class BuriedDeploymentsArray
+        {
+            Consensus _Parent;
+            int[] _Heights;
+            public BuriedDeploymentsArray(Consensus parent)
+            {
+                _Parent = parent;
+                _Heights = new int[Enum.GetValues(typeof(BuriedDeployments)).Length];
+            }
+            public int this[BuriedDeployments index]
+            {
+                get
+                {
+                    return _Heights[(int)index];
+                }
+                set
+                {
+                    _Parent.EnsureNotFrozen();
+                    _Heights[(int)index] = value;
+                }
+            }
+        }
+        public class BIP9DeploymentsArray
+        {
+            Consensus _Parent;
+            BIP9DeploymentsParameters[] _Parameters;
+            public BIP9DeploymentsArray(Consensus parent)
+            {
+                _Parent = parent;
+                _Parameters = new BIP9DeploymentsParameters[Enum.GetValues(typeof(BIP9Deployments)).Length];
+            }
+
+            public BIP9DeploymentsParameters this[BIP9Deployments index]
+            {
+                get
+                {
+                    return _Parameters[(int)index];
+                }
+                set
+                {
+                    _Parent.EnsureNotFrozen();
+                    _Parameters[(int)index] = value;
+                }
+            }
+        }
+
+        public Consensus()
+        {
+            this.buriedDeployments = new BuriedDeploymentsArray(this);
+            this.bIP9Deployments = new BIP9DeploymentsArray(this);
+        }
+
+        private readonly BuriedDeploymentsArray buriedDeployments;
+        public BuriedDeploymentsArray BuriedDeployments => this.buriedDeployments;
+
+
+        private readonly BIP9DeploymentsArray bIP9Deployments;
+        public BIP9DeploymentsArray BIP9Deployments => this.bIP9Deployments;
+
+        int subsidyHalvingInterval;
+        public int SubsidyHalvingInterval
+        {
+            get
+            {
+                return this.subsidyHalvingInterval;
+            }
+            set
+            {
+                EnsureNotFrozen();
+                this.subsidyHalvingInterval = value;
+            }
+        }
+
+        private Func<BlockHeader, uint256> getPoWHash = h => h.GetHash();
```

Comment:
can't we just move this implementation to the public GetPoWHash, we don't need this whatever it is

Reply:
Yes but I am not sure yet how we will change that so preferred not to change functionality.